src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AFSamplers.java

```diff
@@ -0,0 +1,112 @@
+package org.broadinstitute.hellbender.tools.exome.allelefraction;
+
+import org.apache.commons.math3.random.RandomGenerator;
+import org.broadinstitute.hellbender.utils.mcmc.AdaptiveMetropolisSampler;
+import org.broadinstitute.hellbender.utils.mcmc.Sampler;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by davidben on 11/25/15.
+ */
+public class AFSamplers {
+
+    public static final class MeanBiasSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
+        private static final double INITIAL_MEAN_BIAS_STEP_SIZE = 0.01;
+
+        public MeanBiasSampler() {
+            super(INITIAL_MEAN_BIAS_STEP_SIZE, 0, Double.POSITIVE_INFINITY);
+        }
+
+        @Override
+        protected double proposalLogLikelihood(Double proposedMeanBias, AFState state, AFData data) {
+            return AFModel.logLikelihoodWithProposedMeanBias(state, data, proposedMeanBias);
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.meanBias(); }
+    }
+
+    public static final class BiasRateSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
+        private static final double INITIAL_BIAS_RATE_STEP_SIZE = 10;
+        //not strictly needed, but just in case prevents the model from getting trapped in a silly state during burn-in
+        private static final double MAX_BIAS_RATE = 1000;
+
+        public BiasRateSampler() {
+            super(INITIAL_BIAS_RATE_STEP_SIZE, 0, MAX_BIAS_RATE);
+        }
+
+        @Override
+        protected double proposalLogLikelihood(Double proposedBiasRate, AFState state, AFData data) {
+            return AFModel.logLikelihoodWithProposedBiasRate(state, data, proposedBiasRate);
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.biasRate(); }
+    }
+
+    public static final class POutlierSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
+        private static final double INITIAL_OUTLIER_PROBABILITY_STEP_SIZE = 0.01;
+        private static final double MAX_OUTLIER_PROBABILITY = 0.1;
+
+
+        public POutlierSampler() {
+            super(INITIAL_OUTLIER_PROBABILITY_STEP_SIZE, 0, MAX_OUTLIER_PROBABILITY);
+        }
+
+        @Override
+        protected double proposalLogLikelihood(Double proposedPOutlier, AFState state, AFData data) {
+            return AFModel.logLikelihoodWithProposedPOutlier(state, data, proposedPOutlier);
+        }
+
+        @Override
+        protected double currentValue(final AFState state) { return state.pOutlier(); }
+    }
+
+    // to sample the minor fraction we need an AdaptiveMetropolisSampler that knows which segment it samples
+    public static final class perSegmentMinorFractionSampler extends AdaptiveMetropolisSampler<AFState, AFData> {
```

Comment:
capitalize perSegmentMinorFractionSampler. also, this class can be made private.